packages/vue-3/src/VueNodeViewRenderer.ts

```diff
@@ -221,7 +221,12 @@ export function VueNodeViewRenderer(
     if (!(props.editor as Editor).contentComponent) {
       return {}
     }
+    // check for class-component and normalize if neccessary
+    const normalizedComponent = typeof component === 'function' && '__vccOpts' in component
+      // eslint-disable-next-line no-underscore-dangle
+      ? component.__vccOpts as Component
```

Comment:
Looks good to me – just don't have enough Vue experience in this depth to verify this change. Could you give a short explanation? @wagich

Reply:
That matches my understanding as well, see f.ex. [this comment in vue-router.](https://github.com/vuejs/router/blob/main/packages/router/src/navigationGuards.ts#L300)

As I understand it, `__vccOpts` was defined as an internal API for Vue 3 to access the component definition (in Options API style using `props`, `methods`, `data` etc. members) when using a ES6 class as the component. Even tough the API is internal, it looks to be well used by other (even external) components and is stable since 2020 (see [PR for the feature](https://github.com/vuejs/core/pull/821)).

Reply:
LGTM then